### ISSUE TITLE
laser cut scale issues when transfering from inkscape 

### ISSUE THREAD

**Issue by Thingomy** [DoESLiverpool/somebody-should#1039]
In the early days of laser cutting in DoES, my workflow consisted of generate DXF in Inkscape, import into laser cutter software, (experience a string of borked data, repeat several times... not the point) and eventually the data would come in just fine.

As of some point in time, a year or 2 ago, this stopped happening, the objects would consistently import significantly smaller than they were supposed to. I have believed for a while that this was some kind of broken imperial/metric thing, and have started to investigate, the theory is that data in one unit is being incorrectly interpreted as another unit (see Mars Climate Orbiter). The hack via corel draw is messy and unpleasant.

Both laser cutter machines do it for me; every copy of Inkscape I've tried does it. This suggests it's either a version upgrade to Inkscape, that did it, or some setting has been changed on both laser machines. I may try finding an old DXF file that I know worked back in the day, to narrow things down a bit.

To try to work out what's actually going on numerically, I captured some data on Gerald's computer:

A 500mm wide object saved in inkscape loads at 132.292mm wide in the laser software.

(or if a 1mm object exported from inkscape would import as 0.264584 mm)

500/132.292 ~= 3.779518

3.779518 desktop publishing pt ~= 1.33333mm (according to wolfram alpha)

so if inkscape were saving things in mm and the laser software was expecting pt, we would be off only by a simple fraction which hints that this is close, but not quite there.

More work required on this. Does anyone have any other insights, or know anything. I will continue to investigate, but if anyone has any ideas, please let me know.

To fix we either need to nail down the source of the problem and fix it, or at least find an acceptable work around. Or replace the laser software solution with something better.

Can't put a timescale on this, it will take as long as it takes. It’s been like this for well over a year from my recollection.

**Comment by MatthewCroughan** [comment]
In my experience, export from DXF is the inconsistency, things will always import properly since scale is an important and conserved part of the DXF format. 
However, I've noticed that the Inkscape PC has had its settings constantly changed, in terms of its units. Everything in inkscape on the laser cutting pc for Gerald is measured in pixels, sometimes not as someone changes it back, I'm convinced this affects import and export in a way.
the DXF export plugin asks you what unit you want to export as, though I have a feeling that somewhere in the pipeline this is manipulated by the canvas measurement unit.

This is one of the reasons I'm suggesting we use Linux for things where we can, like the label printing/vinyl cutting PC in #990. This may or may not be the reason this is happening, but at least on Linux we could prevent changes from being made to the software, or at least have a system in place that lets us know when settings have been changed. Or have a user registration system that allows people to make new accounts on the system if they want to change default settings. 

TL;DR Make sure your canvas is set to MM, and that your dxf export is set to MM:

![image](https://user-images.githubusercontent.com/26458780/51582373-b378b880-1ec3-11e9-87b5-8f59d90d0407.png)
![image](https://user-images.githubusercontent.com/26458780/51582399-c68b8880-1ec3-11e9-8936-2c5f692df680.png)

**Comment by DoESsean** [comment]
Are these issues still ongoing?

**Comment by magman2112** [comment]
Closing this issue as Gerald now has new software as in #1642, which should eliminate the import issues which were inherent in the old software.
